raster-foundry/raster-foundry
# Allow specifying color ramps using percentage breakpoints

## Overview

Allows the specification of color ramps based on percentage-of-maximum stops, which allows the creation of ramps which do not have equally-spaced color intervals.

Also adds the following color ramps:
- "Magma"
- "Plasma"
- "Inferno"
- White - Blue
- White - Purple 
- Colorblind friendly (Wistia)
- NDVI-specific (under the diverging category)

### Checklist


### Demo

![ndvigradient](https://user-images.githubusercontent.com/447977/31100601-1972bc98-a798-11e7-9f9f-7f58a85e4185.png)

### Notes
The way that this works on the frontend is kind of cheating. The background is that if you specify a color ramp as an array, the tiler will automatically spread it evenly across the full histogram of a raster, whereas if the color ramp is specified as a mapping, then the tiler expects actual raster values. Therefore, In order to specify the correct breakpoints, the frontend would need to know the project histogram before applying the ramp, which it doesn't always, and if the histogram changes the ramp might become inappropriate, which we would need to detect and either automatically fix or warn the user.

To get around this problem, this instead uses the percentage breakpoints to generate array-based ramps which _mimic_ breakpoints. The advantage of this is that all color ramps within the app are still treated as arrays of values, which doesn't require knowledge of the histogram ahead of time, and will still work if the histogram changes. The disadvantage is that it can't provide perfect results for every raster, and some color ramps could cause the array to explode. However, it seems to work pretty well for the existing color ramps.

## Testing Instructions

 * Adjust color ramps for in both the Project Edit screen and for lab nodes and confirm that everything still works the same (except that there are now more histograms).

Closes #2566
